TESTS/mbed_hal/uart_bare_metal_test/uart_test.py

```diff
@@ -0,0 +1,347 @@
+"""
+import time
+import serial
+import sys
+import mbed_lstools
+from subprocess import Popen, PIPE, STDOUT
+import shutil
+from serial import win32
+import argparse
+import ntpath
+
+"""! UART TEST FOR GREEN TEA
+
+  TEST SCENARIO
+  This part of the test is executed on the host (PC).
+ 
+  After startup it will try to communicate via UART with the mbed device.
+  This part of the test waits for the expected message and responds with
+  the same message. 
+  4 test messages should be received. Last message ("PASSED") should inform
+  that on the mbed side also all messages have been received.
+ 
+  To perform this test with serial port baud rate different than default (9600),
+  call host test with appropriate --baudrate parameter.
+  
+  Example usage:
+      uart_test.py -m K64F -m K64F -t GCC_ARM -b 9600
+  This command will run test on K64F platfom using GCC_ARM toolchain and
+  serial connection speed equal to 9600 b/s.
+  
+  For more options:
+      uart_test.py -h
+"""
+
+serial_port = None
+VERBOSE = False
+RETRY_LIMIT = 10
+
+# Listo of expected messages from the mbed device.
+message_list = ['x\0', 
+                'mbed\0',
+                'Green Tea UART bear metal test.\0', 
+                'Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message--.\0',
+                'PASSED\0']
+
+def log_info(msg):
+    print('[INFO]: %s') % msg
+
+def log_recv(msg):
+    if(VERBOSE == True):
+        print('[RECV]: %s') % msg
+
+def log_send(msg):
+    if(VERBOSE == True):
+        print('[SEND]: %s') % msg
+
+def log_fail(msg):
+    print('[FAIL]: %s') % msg
+    if(serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+    sys.exit()
+
+def log_pass(msg):
+    print('[PASS]: %s') % msg
+    if(serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+        
+def create_test_header_file(baudrate):
+    """! Create header file with the specified baud rate.
+    
+    @param baudrate specified transmission baud rate.
+    """
+    file = open('test_config.h','w') 
+    file.write('/* This file has been automatically generated by the python script. */\n')
+    file.write('/* This file contains serial port baud rate. */\n')
+    file.write('#define BAUD_RATE %s\n' % baudrate)
+    file.close()
+
+def run_command(cmd):
+    """! Runs command and prints proc stdout on screen.
+         It is used to build the test.
+         
+    @param cmd command to be executed.
+    @return image path.
+    """
+    build_success = True
+    image_path = str()
+    
+    try:
+        p = Popen(cmd, stdout=PIPE, stderr=STDOUT)
+    except OSError as e:
+        log_fail('Command can not be executed.')
+        sys.exit()
+    
+    output = str()
+    
+    for line in iter(p.stdout.readline, b''):
+        if('ERROR' in line):
+            build_success = False
+        if('Image:' in line):
+            image_path = line[7:]
```

Comment:
More stable API sould be used instead of parsing `stdout`. Use `mbed test --compile --build-data foo` and then parse `foo` JSON file:
```python
build_data = {}
with open('foo') as f:
    build_data = json.load(f)

try:
    image_path = build_data['builds'][0]['bin']
except KeyError:
    image_path = ''  # or raise an exception
```

Reply:
Fixed as suggested.